llvm/lib/TargetParser/RISCVISAInfo.cpp

```diff
@@ -771,9 +772,11 @@ Error RISCVISAInfo::checkDependency() {
       return getIncompatibleError("xwchc", "zcb");
   }
 
-  if (Exts.count("xqcicsr") != 0 && (XLen != 32)) {
-    return getError("'xqcicsr' is only supported for 'rv32'");
-  }
+  for (auto Ext : XqciExts)
+    if (Exts.count(Ext.str()) && (XLen != 32))
+      return createStringError(errc::invalid_argument,
+                               "'" + Twine(Ext) + "'" +
+                                   " is only supported for 'rv32'");
```

Comment:
There aren't any tests for this error - please may you add some in `RISCVISAInfoTest.cpp`?

Reply:
Thanks Sam. I've added a test in RISCVISAInfoTest.cpp